src/main/java/dude/Storage.java

```diff
@@ -0,0 +1,147 @@
+package dude;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Scanner;
+
+import dude.task.Deadline;
+import dude.task.Event;
+import dude.task.Task;
+import dude.task.Todo;
+
+/**
+ * Storage class that controls task storing functionalities of Dude.
+ */
+public class Storage {
+    private static final String directoryPath = "./data";
+    private String filePath = "./data/dude.txt";
+
+    /**
+     * Class constructor.
+     */
+    public Storage() {
+        boolean success = createStorageIfNotExists();
```

Comment:
Instead of just success, would it be better to attribute some sort of name to it?
But this could just be a nit :)

```suggestion
        boolean storageCreationIsSuccesful = createStorageIfNotExists();
```